1082 monochrome cli output

evrose54 opened the pull request:
## ✨ Summary

The GeoIPS CLI has a few commands which can produce output that is colored
differently than most terminal text. We do this via the 'colorama' package, however,
we did not have a setting to disable colored terminal output if wanted. We took in
consideration what other software packages have implemented and decided to reinforce
that standard by implementing a new 'NO_COLOR' environment variable. If set to True,
then GeoIPS will no longer produce colored output via the terminal. Keep in mind
that since this variable commonly used among software packages, setting NO_COLOR to
True may impact packages outside the scope of GeoIPS. For example, when this
variable is enabled, pytest will no longer color its terminal output.

Files modified:

    - docs/source/concepts/configuration/index.rst
    - docs/source/concepts/functionality/command-line/index.rst
    - docs/source/concepts/index.rst
    - geoips/commandline/commandline_interface.py
    - geoips/commandline/geoips_command.py
    - geoips/commandline/geoips_run.py
    - geoips/commandline/geoips_tree.py
    - geoips/filenames/base_paths.py

---

## ✅ Checklist for Reviewer Reference (filled out by PR Author)

- [x] **Tests**: All tests and CI pass (e.g., full, base, extra, etc.)
    - [x] **Full test**: Yes, full test *is passing*.
- [x] **Integration Tests**: Integration tests added/updated for new/modified functionality *OR* no new functionality
- [x] **Other Repos**: I have updated other repositories to handle this change *OR* my package doesn't impact other plugin packages.

---
## 🔗 Related Issues

- **Fixes:** NRLMMD-GEOIPS/geoips#1082

---
## 🧪 Testing Instructions

I ran ``pytest -v /tests/unit_tests/commandline/`` from the top level directory of the GeoIPS package. Since the changes made here impact nothing but the CLI, I didn't see a need to run all of the integration tests. I can still do that if you'd like though.

evrose54 committed:
added logic to turn off color for the CLI

evrose54 committed:
fixed small bug

evrose54 committed:
added comment explaining potential effects of setting this new environment variable

evrose54 committed:
added release notes

evrose54 committed:
added documentation on the 'NO_COLOR' environment variable

evrose54 committed:
updated release note for newly modified files

biosafetylvl5 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

evrose54 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

biosafetylvl5 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

evrose54 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

biosafetylvl5 committed:
Change default value of NO_COLOR to False

biosafetylvl5 committed:
Update output directory default to use PATHS

biosafetylvl5 commented on geoips/filenames/base_paths.py:
```diff
@@ -111,6 +133,10 @@ def initialize_paths():
+    paths["NO_COLOR"] = get_env_var(
+        "NO_COLOR",
+        True,
+    )
```
I changed this to False

biosafetylvl5 committed:
Apply black

biosafetylvl5 committed:
Revise GeoIPS configuration documentation

Updated the configuration documentation for GeoIPS, including environment variable descriptions and examples for setting them in various shells.

biosafetylvl5 reviewed (dismissed):
Approving with the condition the docs get updated :)

evrose54 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

evrose54 committed:
Merge branch '1082-monochrome-CLI-output' into no-color-documentation

biosafetylvl5 committed:
Fix formatting of environment variable sections in docs

evrose54 committed:
Revise GeoIPS configuration documentation (#1241)

biosafetylvl5 committed:
Update documentation with correct new lines

biosafetylvl5 committed:
Merge branch 'main' into 1082-monochrome-CLI-output

evrose54 committed:
Merge branch 'main' into 1082-monochrome-CLI-output
